app/presenters/vote_button_presenter.rb

```diff
@@ -1,9 +1,71 @@
 class VoteButtonPresenter < ApplicationPresenter
-  def vote?
-    parents_vote.blank?
+  presents :parent
+
+  def vote
+    @vote ||= Vote.find_by(user_id: h.current_user, voteable_id: parent)
```

Comment:
I dig the memoization 👏 , but this is still going to run a query if no vote exists and `@vote` is set to `nil`. A pattern to get around this is:

```ruby
def vote
  return @vote if defined?(@vote)

  @vote = Vote.find_by(user_id: h.current_user.id, voteable_id: parent.id)
end
```

This ensures that if `@vote` has been set to `nil`, then that `nil` value is returned and the SQL query in `find_by` won't run again. Does that make sense?

Reply:
yep!